### PR TITLE
Bunch of e2e tests for making sure limit handling is correct in VPA

### PR DESCRIPTION
**Pull request by bskiba**

### CODE REVIEW

File: vertical-pod-autoscaler/e2e/v1beta1/admission_controller.go [hunk 1]
```diff
@@ -88,6 +90,119 @@ var _ = AdmissionControllerE2eDescribe("Admission-controller", func() {
...
+		ginkgo.By("Setting up a hamster deployment")
+		podList := startDeploymentPods(f, d)
+
+		// Originally Pods had 100m CPU, 100Mi of memory, but admission controller
+		// should change it to 200m CPU (as this is the recommendation
+		// capped according to max limit in LimitRange) and 200Mi of memory,
+		// which is uncapped. Limit to request ratio should stay unchanged.
+		for _, pod := range podList.Items {
+			gomega.Expect(*pod.Spec.Containers[0].Resources.Requests.Cpu()).To(gomega.Equal(ParseQuantityOrDie("200m")))
+			gomega.Expect(*pod.Spec.Containers[0].Resources.Requests.Memory()).To(gomega.Equal(ParseQuantityOrDie("200Mi")))
```
Comment (jbartosik) [inline vertical-pod-autoscaler/e2e/v1beta1/admission_controller.go]: Why not check limit <= max limit?

Comment (bskiba) [reply vertical-pod-autoscaler/e2e/v1beta1/admission_controller.go]: Done

**Pull request merged**